pandas-stubs/tseries/holiday.pyi

```diff
@@ -1,3 +1,148 @@
-class AbstractHolidayCalendar: ...
+from collections.abc import Callable
+from datetime import (
+    date as _date,
+    datetime,
+)
+from typing import overload, Literal
+
+import numpy as np
+from pandas import (
+    DatetimeIndex,
+    Series,
+)
+from pandas._libs.tslibs.offsets import BaseOffset
+from pandas._libs.tslibs.timestamps import Timestamp
+
+def next_monday(dt: datetime) -> datetime: ...
+def next_monday_or_tuesday(dt: datetime) -> datetime: ...
+def previous_friday(dt: datetime) -> datetime: ...
+def sunday_to_monday(dt: datetime) -> datetime: ...
+def weekend_to_monday(dt: datetime) -> datetime: ...
+def nearest_workday(dt: datetime) -> datetime: ...
+def next_workday(dt: datetime) -> datetime: ...
+def previous_workday(dt: datetime) -> datetime: ...
+def before_nearest_workday(dt: datetime) -> datetime: ...
+def after_nearest_workday(dt: datetime) -> datetime: ...
+
+class Holiday:
+    @overload
+    def __init__(
+        self,
+        name: str,
+        year: int | None = ...,
+        month: int | None = ...,
+        day: int | None = ...,
+        offset: BaseOffset | list[BaseOffset] = ...,
+        observance: None = ...,
+        # Values accepted by Timestamp(), or None:
+        start_date: (
+            np.integer | float | str | _date | datetime | np.datetime64 | None
+        ) = ...,
+        end_date: (
+            np.integer | float | str | _date | datetime | np.datetime64 | None
+        ) = ...,
+        days_of_week: tuple[int, ...] | None = ...,
+    ) -> None: ...
+    @overload
+    def __init__(
+        self,
+        name: str,
+        year: int | None = ...,
+        month: int | None = ...,
+        day: int | None = ...,
+        offset: None = ...,
+        observance: Callable[[datetime], datetime] = ...,
+        # Values accepted by Timestamp(), or None:
+        start_date: (
+            np.integer | float | str | _date | datetime | np.datetime64 | None
+        ) = ...,
+        end_date: (
+            np.integer | float | str | _date | datetime | np.datetime64 | None
+        ) = ...,
+        days_of_week: tuple[int, ...] | None = ...,
+    ) -> None: ...
+    @overload
+    def dates(
+        self,
+        start_date: (
+            np.integer | float | str | _date | datetime | np.datetime64 | None
+        ),
+        end_date: (np.integer | float | str | _date | datetime | np.datetime64 | None),
+        return_name: Literal[False] = ...,
+    ) -> DatetimeIndex: ...
+    @overload
+    def dates(
+        self,
+        start_date: (
+            np.integer | float | str | _date | datetime | np.datetime64 | None
+        ),
+        end_date: (np.integer | float | str | _date | datetime | np.datetime64 | None),
+        return_name: Literal[True] = ...,
+    ) -> Series: ...
+
+holiday_calendars: dict[str, type["AbstractHolidayCalendar"]]
+
+def register(cls: type["AbstractHolidayCalendar"]) -> None: ...
+def get_calendar(name: str) -> "AbstractHolidayCalendar": ...
```

Comment:
If you put `from __future__ import annotations` at the top of this file, then you can remove the quotes around `"AbstractHolidayCalendar"` in this line and the above, as well as in places below.